lms/static/js/student_account/views/account_settings_factory.js

```diff
@@ -36,29 +42,94 @@
             userPreferencesModel = new UserPreferencesModel();
             userPreferencesModel.url = userPreferencesApiUrl;
 
-            if (allowEmailChange) {
+            if (syncLearnerProfileData && enterpriseName) {
```

Comment:
Have you written/updated the Jasmine tests for these changes ?

Reply:
I am updating the jaamine tests.